manager/recurringjob.go

```diff
@@ -114,9 +114,64 @@ func (m *VolumeManager) UpdateRecurringJob(spec longhorn.RecurringJobSpec) (*lon
 }
 
 func (m *VolumeManager) DeleteRecurringJob(name string) error {
+	if err := m.CleanVolumeRecurringJob(name); err != nil {
+		return err
+	}
+
 	if err := m.ds.DeleteRecurringJob(name); err != nil {
 		return err
 	}
 	logrus.Infof("Deleted recurring job %v", name)
 	return nil
 }
+
+func (m *VolumeManager) CleanVolumeRecurringJob(name string) error {
+	recurringJob, err := m.ds.GetRecurringJob(name)
+	if err != nil {
+		return err
+	}
+
+	// check if other recurring jobs contains same groups
+	recurringJobs, err := m.ds.ListRecurringJobs()
+	if err != nil {
+		return err
+	}
+	rmGroups := []string{}
+	for _, group := range recurringJob.Spec.Groups {
+		inUse := 0
+		for _, obj := range recurringJobs {
+			if util.Contains(obj.Spec.Groups, group) {
+				inUse++
+			}
+			if inUse > 1 {
+				continue
+			}
+		}
+		if inUse == 1 {
```

Comment:
Could we use `inUse == 0` as the condition by excluding the job which is `recurringJob`? Using `==1` is not intuitive or implicit.

Reply:
Updated to use boolean instead.